src/EnergyPlus/SolarShading.cc

```diff
@@ -8190,6 +8160,7 @@ namespace SolarShading {
                         if (SurfWinOriginalClass(SurfNum) == SurfaceClass_TDD_Diffuser) {
                             PipeNum = SurfWinTDDPipeNum(SurfNum);
                             SurfNum2 = TDDPipe(PipeNum).Dome;
+                            CosInc = CosIncAng(TimeStep, HourOfDay, SurfNum2);
```

Comment:
Seems like `CosInc` should be declared here with local scope (and also up in line 6360?). Possibly the same for some of these other variables. If you are planning that already in the next windowstruct branch, then it's ok here as-is.